### PR TITLE
LIVE-1740: Update types to 4.0.0

### PR DESCRIPTION
**Pull request by dskamiotis**
## Why are you doing this?

Adds `Letter` and `Labs` types. 

Currently parked waiting on the following:
- Image-Rendering types bump to 4.0.0
- changes to `relatedItemTypes` thrift https://github.com/guardian/apps-rendering/pull/1180

## Changes

- Upgraded to Types 4.0.0
- Changed use of `Design.AdvertisementFeature` to `Special.Labs`
- Used `LABS` in `relatedItemType`

## Screenshots

| Before | After |
| --- | --- |
|  |  |

### CODE REVIEW

File: src/capi.ts [hunk 1]
```diff
@@ -13,7 +13,7 @@ import { fromString as dateFromString } from 'date';
 import type { MainMedia } from 'headerMedia';
 import { MainMediaKind } from 'headerMedia';
 import { parseImage } from 'image';
-import { isAdvertisementFeature } from 'item';
+import { isSpecialLabs } from 'item';
```
Comment (joecowton1) [inline src/capi.ts:16]: We should make sure everything is referring to `Labs` rather than `SpecialLabs` throughout.

ie this should be `isLabs`

**Review by joecowton1 — COMMENTED** [review]
Good work @dskamiotis .

A few things we're missing here:
- We should make sure everything is referring to `Labs` rather than `SpecialLabs` throughout.
- There's a lot of linting errors that need resolving
- Can we have a comment explaining  the current status of this PR? It makes it easier for anyone else coming across this PR to understand what's going on. It would be good to define what needs doing still and why there is a 'do not merge' etc.

**Comment by joecowton1** [comment]
@JamieB-gu have spent sometime with @dskamiotis on this both today and Friday, we've done the grunt work in terms of updating the `@guardian/types` package to `4.0.0` and replacing references to `Design.AdvertisingFeature` with `Special.Labs` , but hit a thorny patch of errors mostly in the style of:

```
TS2322: Type 'import("/Users/code/apps-rendering/node_modules/@guardian/types/dist/format").Format' is not assignable to type 'import("/Users/code/apps-rendering/node_modules/@guardian/image-rendering/node_modules/@guardian/types/dist/format").Format'.
```

I'm not sure whether this is due to discrepancies in Typescript versions or similar or what really, but we hoped with your expertise you might be able to work this one out.

** Upgrading the `@guardian/types` package also requires an update to add labs to the `relatedItemType` file within the `apps-rendering-api-models` module. @marjisound has made a PR for that here: [link when ready]

**Comment by joecowton1** [comment]
@dskamiotis pulled this down this morning to check it was working and in merging master back in yesterday you had reinstated the original `AdvertisementFeature` folder which was causing a load more type errors. I've removed it and push the commit so its fixed now, but please be careful when merging things and check that they work before pushing 👍

**Comment by JamieB-gu** [comment]
This is no longer blocked on changes to `relatedItemTypes` (see #1180), or the image-rendering changes (https://github.com/guardian/image-rendering/pull/227). Once #1197 has gone out and we've upgrade to Source v3, we can bump image-rendering and ship this PR.

**Comment by faresite** [comment]
This was done in https://github.com/guardian/apps-rendering/pull/1244

**Pull request closed without merging**